chunky/src/java/se/llbit/chunky/world/region/RegionParser.java

```diff
@@ -60,7 +58,7 @@ public RegionParser(WorldMapLoader loader, RegionQueue queue, MapView mapView) {
         }
         ChunkView map = mapView.getMapView();
         if (map.isRegionVisible(position)) {
-          Dimension dimension = mapLoader.getWorld().currentDimension();
+          JavaDimension dimension = (JavaDimension) mapLoader.getWorld().currentDimension(); // FIXME: don't cast
```

Comment:
what's still only in `JavaDimension` that is needed here? That might show what the `Dimension` class is still missing

Reply:
Dimension doesn't have regions. the map view is very region based.
Maybe we should simulate regions on every world to cope with this as you previously suggested